File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> Final; 
        for(int i = 0; i < nums1.size(); i++){
            for(int j = 0; j < nums2.size(); j++){
                if(nums1[i] == nums2[j]){
                    if(j == nums2.size() - 1){
                        Final.push_back(-1); 
                    }

                    for(int k = j + 1; k < nums2.size(); k++){
                        if(nums2[k] > nums2[j]){
                            Final.push_back(nums2[k]); 
                            break; 
                        }
                        if(k == nums2.size() - 1) {
                            Final.push_back(-1); 
                        }
                        
                    
                    }
                }
            }
        }
        return Final; 
        
    }
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        ans = [-1]*len(nums2)
        mapp = defaultdict(int)
        stk = []
        for i in range(len(nums2)-1, -1,-1): 
            while stk and nums2[i] >= stk[-1]: 
                stk.pop()
            
            if stk: 
                ans[i] = stk[-1]
            mapp[nums2[i]] = ans[i]
            
            stk.append(nums2[i])
        
        for i in range(len(nums1)): 
            nums1[i] = mapp[nums1[i]]
        
        return nums1
